ribs/archives/_cvt_archive.py

```diff
@@ -260,12 +264,30 @@ def index_of(self, measures_batch):
             _, indices = self._centroid_kd_tree.query(measures_batch)
             return indices.astype(np.int32)
 
+        expanded_dims = np.expand_dims(measures_batch, axis=1)
+        if self.chunk_distances:  # Compute indices chunks at a time
+            distances = np.sum(np.square(
+                expanded_dims[0:min(len(measures_batch), self.chunk_size)] -
+                self.centroids),
+                               axis=2)
+            indices = np.argmin(distances, axis=1).astype(np.int32)
+            for i in range(
+                    2,
+                    len(measures_batch) // self.chunk_size +
+                    len(measures_batch) % self.chunk_size + 1):
+                l = (i - 1) * self.chunk_size
+                r = min(len(measures_batch), i * self.chunk_size)
+                current_batch = expanded_dims[l:r] - self.centroids
+                current_res = np.argmin(np.sum(np.square(current_batch),
+                                               axis=2),
+                                        axis=1).astype(np.int32)
+                indices = np.concatenate((indices, current_res))
```

Comment:
This is incurring O(n^2) runtime because you repeatedly concatenate indices with the new res and thus create a new array. Can you make a list that stores all the `current_res`, then just do a single concatenation at the end into `indices`?

Reply:
Great idea, on it